### DISCUSSION TITLE
@micropython.asm_thumb error

### DISCUSSION THREAD

**Discussion by GKellu** [micropython/micropython#11006] [RP2040 / Pico]
Hi everyone, I try to use following code on pi pico with micropython v1.19.1:

@micropython.asm_thumb
def asm_add(r0, r1):
    add(r0, r0, r1)

Code is from latest documentation on micropython. I get this error:

Traceback (most recent call last):
  File "<stdin>", line 3, in asm_add
SyntaxError: expecting an assembler instruction

I get this error on every code tested from documentation. I have tested it on  micropython v1.17 with same result. 
Please help me how to use inline assembler on pi pico with micropython.

**Comment by robert-hh** [comment]
Maybe you try the code with the actual nightly build of the firmware. If I run your example with that, I get no error.
```
@micropython.asm_thumb
def asm_add(r0, r1):
    add(r0, r0, r1)

print(asm_add(1,11))
```

**Reply by GKellu** [reply]
Hi Robert, thank you for your answer but it did not worked, I have downloaded actual nightfly uf2, put it on the pico and run the code. This is the result copied from Thony IDE shell window:

MicroPython v1.19.1-963-g668a7bd28 on 2023-03-10; Raspberry Pi Pico with RP2040

Type "help()" for more information.
>>> @micropython.asm_thumb
def asm_add(r0, r1):
    add(r0, r0, r1)

print(asm_add(1,11))
Traceback (most recent call last):
  File "<stdin>", line 3, in asm_add
SyntaxError: expecting an assembler instruction
>>>

**Reply by robert-hh** [reply]
Try to use it without Thonny, like with a simple terminal emulator. Or put the script in a local file like asmtest.py and run:

`mpremote run asmtest.py`

Thonny has quite a bit of magic under the hood, which sometimes is helpful, sometimes a pain. So if anything strange happens when using Thonny, try without Thonny.

**Reply by robert-hh** [reply]
Please use proper formatting when posting code. You can get that by enclosing the code with lines containing three backticks: ```

**Reply by GKellu** [reply]
Thank you Robert, The problem is Thony IDE. Can you recommend other free IDE to use with Windows?

**Reply by robert-hh** [reply]
Actually not, sorry. I do not use IDEs with Micropython boards. I just use a local Editor and simple tools like mpremote, picocom, ftp.

**Reply by GKellu** [reply]
Thank you.
